# HG changeset patch
# User Ray Donnelly <mingw.android@gmail.com>
# Date 1471216218 -3600
#      Mon Aug 15 00:10:18 2016 +0100
# Node ID 5b0b4f32f6f9e1c0a6d023dc6d066474bd87d2de
# Parent  1792d96eaee85143f4426bbf5a8c21ac0f9b3bc2
Win32: Change R_USER check for HOMEDIR to HOMEDRIVE

.. also do not de-reference HOMEPATH if it's a NULL pointer.

diff --git a/rpy/rinterface/_rinterface.c b/rpy/rinterface/_rinterface.c
--- a/rpy/rinterface/_rinterface.c
+++ b/rpy/rinterface/_rinterface.c
@@ -1262,8 +1262,8 @@
     strcpy(RUser, getenv("R_USER"));
   } else if (getenv("HOME")) {
     strcpy(RUser, getenv("HOME"));
-  } else if (getenv("HOMEDIR")) {
-    strcpy(RUser, getenv("HOMEDIR"));
+  } else if (getenv("HOMEDRIVE") && getenv("HOMEPATH")) {
+    strcpy(RUser, getenv("HOMEDRIVE"));
     strcat(RUser, getenv("HOMEPATH"));
   } else {
     PyErr_Format(PyExc_RuntimeError, "R_USER not defined.");
